sklearn/model_selection/_search.py

```diff
@@ -999,3 +1004,356 @@ def fit(self, X, y=None, labels=None):
+            Target relative to X for classification or regression;
+            None for unsupervised learning.
+
+        labels : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+        """
+        cv_scores, grid_scores, tested_parameters = [], [], []
+        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
+        cv = check_cv(self.cv, X, classifier=is_classifier(self.estimator))
+
+        #  Initialize with random candidates
+        init_candidates = ParameterSampler(
+            self.param_distributions, self.n_init,
+            random_state=self.random_state)
+
+        for i, params in enumerate(init_candidates):
+            score, all_scores = self._evaluate_params(
+                params, cv, X, y, labels)
+
+            if self.verbose > 0:
+                print('Step ' + str(i) + ' - Hyperparameter '
+                      + str(params) + ', score: ' + str(score))
+            tested_parameters.append(list(params.values()))
+            cv_scores.append(score)
+            grid_scores.append(
+                _CVScoreTuple(params, score, np.array(all_scores)))
+
+        for i in range(self.n_iter-self.n_init):
+            # Model with a Gaussian Process
+            gp = GaussianProcessRegressor(**self.gp_params)
+            gp.fit(tested_parameters, cv_scores)
+
+            # Sample candidates and predict their corresponding
+            # acquisition values
+            candidates = ParameterSampler(
+                self.param_distributions, self.n_candidates,
+                random_state=self.random_state)
+            candidate_values = []
+            candidate_dict = []
+            for s in candidates:
+                # get the candidates as an array
+                candidate_values.append(list(s.values()))
+                candidate_dict.append(s)
+
+            if self.acquisition_function == 'UCB':
+                predictions, std = gp.predict(
+                    candidate_values, return_std=True)
+                acquisition_func = predictions + self.kappa * std
+            elif self.acquisition_function == 'EI':
+                predictions, std = gp.predict(
+                    candidate_values, return_std=True)
+                y_best = np.max(cv_scores)
+                Z = (predictions - y_best - self.xi) / std
+                acquisition_func = std * (
+                    Z * stats.norm.cdf(Z) + stats.norm.pdf(Z))
+                # best_candidate = candidates[np.argmax(ei)]
+            else:
+                raise ValueError(
+                    'acquisition_function not implemented yet : '
+                    + self.acquisition_function)
+            best_idx = np.argmax(acquisition_func)
+            best_candidate = candidate_dict[best_idx]
+            score, all_scores = self._evaluate_params(
```

Comment:
Is there any advantage in considering an acquisition function that generates *batches* of parameter settings to test concurrently before deciding the next? (Certainly this seems a reasonable thing to do initially.) Should we model this as evaluating a sequence of parameter setting batches instead of individual settings? (Perhaps doing so also entails more overlap with `BaseSearchCV`.)